### PR TITLE
FIX: Support unicode in search filter @username

### PR DESCRIPTION
**Pull request by dujiajun**
Fix the bug mentioned in https://meta.discourse.org/t/no-search-result-if-filtered-by-unicode-username/243349

### CODE REVIEW

**Comment by SamSaffron** [comment]
Isn't .* going to catch spaces, eg: `@something something something`, I thought we already had a username regex, can we not use it? 

https://github.com/discourse/discourse/blob/7f01e67e1f0052aeed5bb56f0033428bfa2afeb8/app/models/site_setting.rb#L135-L135

**Comment by dujiajun** [comment]
`SiteSetting.allowed_unicode_username_characters_regex` only include the non-ascii part. So I need to concatenate it with the ascii characters. In this way, I need to write the regex like "^\@([a-zA-Z0-9_\-.#{SiteSetting.unicode_usernames ?SiteSetting.allowed_unicode_username_characters :""}]+)$". However, this will break the unitest because `SiteSetting.unicode_usernames` is loaded before set in the test, and every change to allowed_unicode_username_characters need a restart to take effect. I have no idea about how to deal with this.

**Comment by dujiajun** [comment]
@SamSaffron @gschlager, hello sam and gerhard, I change the regex to (\S+) and it seems to meet your requirements. I believe there is no need to use SiteSetting.allowed_unicode_username_characters_regex.

File: lib/search.rb [hunk 1]
```diff
@@ -646,7 +646,7 @@ def post_action_type_filter(posts, post_action_type)
     end
   end
 
-  advanced_filter(/^\@([a-zA-Z0-9_\-.]+)$/i) do |posts, match|
+  advanced_filter(/^\@(\S+)$/i) do |posts, match|
     username = match.downcase
 
     user_id = User.where(staged: false).where(username_lower: username).pluck_first(:id)
```
Comment (gschlager) [inline lib/search.rb:652]: The username should be normalized in order to fully work with Unicode.

```suggestion
    username = User.normalize_username(match)

    user_id = User.not_staged.where(username_lower: username).pluck_first(:id)
```

**Review by gschlager — CHANGES_REQUESTED** [review]

**Review by gschlager — APPROVED** [review]
Looks good, thank you!

**Pull request merged**